usecase/pokemon.go

```diff
@@ -0,0 +1,63 @@
+package usecase
+
+import (
+	"net/http"
+	"pokeapi/model"
+	"pokeapi/service"
+)
+
+type Usecase struct {
+	csvService service.NewCsvService
+}
+
+type IUsecase interface {
+	GetPokemons() ([]model.Pokemon, *model.Error)
+	GetPokemon(pokemonId int) (model.Pokemon, *model.Error)
+	GetPokemonsFromExternalAPI(newPokemons *[]model.SinglePokeExternal) *model.Error
+}
+
+func New(s service.NewCsvService) *Usecase {
+	return &Usecase{s}
+}
+
+func (us *Usecase) GetPokemons() ([]model.Pokemon, *model.Error) {
+	const pathFile = "./csv/pokemon.csv"
+	f, err := us.csvService.Open(pathFile)
+
+	if err != nil {
+		return nil, &model.Error{
+			Message: err.Error(),
+			Code:    http.StatusInternalServerError,
+		}
+	}
+	return us.csvService.GetPokemons(f)
+}
+
+func (us *Usecase) GetPokemon(pokemonId int) (model.Pokemon, *model.Error) {
+	const pathFile = "./csv/pokemon.csv"
+	f, err := us.csvService.Open(pathFile)
+
+	if err != nil {
+		return model.Pokemon{}, &model.Error{
+			Message: err.Error(),
+			Code:    http.StatusInternalServerError,
+		}
+	}
+	return us.csvService.GetPokemon(pokemonId, f)
+}
+
+func (us *Usecase) GetPokemonsFromExternalAPI(newPokes *[]model.SinglePokeExternal) *model.Error {
```

Comment:
What does this function is supposed to do? The logic inside doesn't seems to correspond to what it's actually doing

Reply:
Removed as refactored from the comment about to don't open files in usecase